overlord/snapstate/autorefresh.go

```diff
@@ -489,21 +493,53 @@ func getTime(st *state.State, timeKey string) (time.Time, error) {
 	return t1, nil
 }
 
+func populateRefreshHints(refreshInfo *userclient.PendingSnapRefreshInfo, snapInfo *snap.Info, err *BusySnapError) {
+	for _, appName := range err.AppNames() {
+		if app, ok := snapInfo.Apps[appName]; ok {
+			path := app.DesktopFile()
+			if _, err := os.Stat(path); err == nil {
+				refreshInfo.BusyAppName = appName
+				refreshInfo.BusyAppDesktopEntry = strings.SplitN(filepath.Base(path), ".", 2)[0]
```

Comment:
is this robust enough because DesktopFile will always end in ".desktop" and we don't allow "." in the names of desktop files?

Reply:
The desktop file is generated by us so yes. I think this is sensible.